include/seqan3/alignment/configuration/global_config.hpp

```diff
@@ -0,0 +1,121 @@
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Provides seqan3::align::global_config
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/core/algorithm/all.hpp>
+#include <seqan3/core/metafunction/basic.hpp>
+
+namespace seqan3::detail
+{
+
+// ----------------------------------------------------------------------------
+// global_alignment_cfg
+// ----------------------------------------------------------------------------
+
+//!\brief The global_alignment_configuration element to distinguish between alignment policies.
+class global_alignment_cfg : public config_element_base<global_alignment_cfg>
+{
+    //!\brief Grant access to private state.
+    friend class config_element_access<global_alignment_cfg>;
+
+    //!\brief State is ignored.
+    remove_cvref_t<decltype(std::ignore)> state;
+};
+}  // namespace seqan3::detail
+
+namespace seqan3::align
+{
+
+/*!\brief Global alignment configuration object.
+ * \ingroup align
+ */
+template <seqan3::detail::config_element_concept ... cfg_elements_t>
+class global_config : public detail::configuration<cfg_elements_t...>
+{
+    //!\brief The base type.
+    using base_type = detail::configuration<cfg_elements_t...>;
+
+public:
+    /*!\name Constructor, destructor and assignment
+     * \{
+     */
+    global_config()                                  = delete;
+    global_config(global_config const &)             = default;
+    global_config(global_config &&)                  = default;
+    global_config & operator=(global_config const &) = default;
+    global_config & operator=(global_config &&)      = default;
+    ~global_config()                                 = default;
+
+    //TODO(rrahn): Using the intermediate concept form here causes an ICE!
+    //!\brief Construction from another seqan3::detail::configuration object.
+    template <typename ... other_configs_t>
+    //!\cond
+        requires (detail::config_element_concept<other_configs_t> && ...)
+    //!\endcond
+    global_config(detail::configuration<other_configs_t...> const & other) : base_type{other}
+    {}
+
+    //TODO(rrahn): Using the intermediate concept form here causes an ICE!
+    //!\brief Construction from another seqan3::detail::configuration object.
+    template <typename ... other_configs_t>
+    //!\cond
+        requires (detail::config_element_concept<other_configs_t> && ...)
```

Comment:
why don't you just enforce this concept in `detail::config_element_concept` so you don't need to do it here everywhere?

Reply:
Can you please elaborate?

Reply:
I meant in `detail::configuration` sorry.